File: 0134-gas-station/0134-gas-station.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        int currentPetrol = 0, prevPetrol = 0, startIndex = 0;
        for (int i = 0; i < ((int) gas.size()); ++i) {
            currentPetrol += (gas[i] - cost[i]);
            if (currentPetrol < 0) {
                startIndex = i + 1;
                prevPetrol += currentPetrol;
                currentPetrol = 0;
            }
        }
        return (currentPetrol + prevPetrol >= 0) ? startIndex : -1;
    }
};

File: 0134-gas-station/0134-gas-station.java
class Solution {
    public static int canCompleteCircuit(int[] gas, int[] cost) {
        int n = gas.length;
        int[] arr = new int[n];
        for (int i = 0; i < n; ++i)
            arr[i] = gas[i] - cost[i];
        for (int i = 0; i < n; ++i) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
        int totalGas = 0, maxCost = 0, startIndex = 0;
        for (int i=0; i<n; ++i) {
            totalGas += arr[i];
            if (totalGas < 0) {
                startIndex = i + 1;
                maxCost += Math.abs(totalGas);
                totalGas = 0;
            }
        }
        return (totalGas >= maxCost) ? startIndex : -1;
    }
}